vllm/entrypoints/openai/api_server.py

```diff
@@ -106,16 +106,20 @@ async def build_async_engine_client(args) -> AsyncIterator[AsyncEngineClient]:
 
     # Otherwise, use the multiprocessing AsyncLLMEngine.
     else:
+        # Select random path for IPC.
+        rpc_path = get_open_zmq_ipc_path()
+        logger.info("Multiprocessing frontend to use %s for RPC Path.",
+                    rpc_path)
```

Comment:
I think you need to bind the socket here, get the path, and pass the path to `rpc_server_process`, while passing the socket directly to `AsyncEngineRPCClient` .

Reply:
I cannot bind the socket here, the binding has to happen inside the server process